x-pack/plugins/stack_alerts/public/alert_types/es_query/validation.ts

```diff
@@ -92,5 +101,20 @@ export const validateExpression = (alertParams: EsQueryAlertParams): ValidationR
       })
     );
   }
+  if (!size) {
+    errors.size.push(
+      i18n.translate('xpack.stackAlerts.esQuery.ui.validation.error.requiredSizeText', {
+        defaultMessage: 'Size is required.',
+      })
+    );
+  }
+  if ((size && size < 0) || size > 10000) {
```

Comment:
I tried putting the `ES_QUERY_MAX_HITS_PER_EXECUTION` const into `common` and using it inside this validation function but I got an optimizer error related to joi.

Reply:
I think this is caused by a `config.ts` file that shouldn't be in `common` (https://github.com/elastic/kibana/blob/master/x-pack/plugins/stack_alerts/common/config.ts). If that file gets moved to the server-side, it could work.